onboarding.md

```diff
@@ -0,0 +1,591 @@
+
+1. add the config value to the `config.properties` file
+2. access and read the config value from code
+
+#### 1. Add the config value
+
+Simply add a new line with an arbitrary key to your `config.properties`:
+
+```properties
+edc.samples.03.logprefix=MyLogPrefix
+```
+
+#### 2. Access the config value
+
+The `ServiceExtensionContext` exposes a method `getSettings(String, String)` to read settings (i.e. config values)'.
+Modify the code from the `HealthEndpointExtension.java` as shown below (use the one from the `samples/03-configuration`
+of course):
+
+```java
+public class HealthEndpointExtension implements ServiceExtension {
+    private static final String LOG_PREFIX_SETTING = "edc.samples.03.logprefix"; // this constant is new
+
+    @Override
+    public Set<String> requires() {
+        return Set.of("edc:webservice");
+    }
+
+    @Override
+    public void initialize(ServiceExtensionContext context) {
+        var logPrefix = context.getSetting(LOG_PREFIX_SETTING, "health"); //this line is new
+        var webService = context.getService(WebService.class);
+        webService.registerController(new HealthApiController(context.getMonitor(), logPrefix));
+    }
+}
+```
+
+Next, we must modify the constructor signature of the `HealthApiController` class and store the `logPrefix` as variable:
+
+```java
+
+@Consumes({MediaType.APPLICATION_JSON})
+@Produces({MediaType.APPLICATION_JSON})
+@Path("/")
+public class HealthApiController {
+
+    private final Monitor monitor;
+    private final String logPrefix;
+
+    public HealthApiController(Monitor monitor, String logPrefix) {
+        this.monitor = monitor;
+        this.logPrefix = logPrefix;
+    }
+
+    @GET
+    @Path("health")
+    public String checkHealth() {
+        monitor.info(String.format("%s :: Received a health request", logPrefix));
+        return "I'm alive!";
+    }
+}
+```
+
+There are a few things worth mentioning here:
+
+- things like configuration value names should be implemented as constants, e.g. `LOG_PREFIX_SETTING` and should have a
+  consistent and hierarchical naming scheme
+- if a config value is not present, we should either specify a default value (i.e. `"health"`) or throw
+  an `EdcException`
+- configuration values should be handled in the `*Extension` class, as it's job is to set up the extension and its
+  required business logic (e.g. the controller). The extension itself should not contain any business logic.
+- it's better to pass the config value directly into the business logic than passing the
+  entire `ServiceExtensionContext`
+
+> _The complete sample code for this chapter is in `samples/03-configuration`._
+
+## Implement a simple file transfer
+
+Now this is a big one:
+
+- We'll create an additional connector, so that in the end we have two connectors, a consumer and a provider.
+- The connectors will talk to each other via IDS, and the consumer will expose a REST API so that external systems (e.g.
```

Comment:
IDS protocol is something we will provide during the next weeks, right now it's just a message quite similar to ArtifactRequest, or?

Reply:
at this time, we only use a few IDS messages (Query, ArtifactRequest). I improved the text accordingly.